### PR TITLE
Make the Create Job page

### PR DESCRIPTION
**Pull request by lotif**
# PR Type
Feature

# Short Description

Clickup Ticket(s): https://app.clickup.com/t/86884uxcu

Making the "New Job" page:
Screenshot 2024-05-27 at 22 44 34

Some additional things:
- In the code, I'm prefixing all the components with `EditJob` because I plan to modify the same page to enable editing jobs
- Adding a `New Job` button to the list jobs page
- Creating 2 server endpoints to retrieve the list of available models and clients, and also adding their react hooks
- Adding a react hook for posting to an endpoint
- Adding `immer` dependency to help with page state management
- Adding `florist.css` with customizations to the Material Dashboard styles

# Tests Added
`florist/tests/integration/api/test_server.py`
`florist/tests/unit/app/jobs/edit/page.test.tsx`

### CODE REVIEW

File: florist/api/server.py [hunk 1]
```diff
@@ -37,3 +40,23 @@ async def lifespan(app: FastAPI) -> AsyncGenerator[Any, Any]:
 app.include_router(training_router, tags=["training"], prefix="/api/server/training")
 app.include_router(job_router, tags=["job"], prefix="/api/server/job")
 app.include_router(status_router, tags=["status"], prefix="/api/server/check_status")
+
+
+@app.get(path="/api/server/models", response_description="Returns a list of all available models")
+def list_models() -> JSONResponse:
+    """
+    Return a list all available models.
```
Comment (jewelltaylor) [inline florist/api/server.py:48]: nit: list of all

File: florist/app/jobs/hooks.tsx [hunk 2]
```diff
@@ -8,3 +10,40 @@ export function useGetJobsByJobStatus(status: string) {
...
+
+export function useGetModels() {
+    return useSWR("/api/server/models", fetcher);
+}
+
+export function useGetClients() {
+    return useSWR("/api/server/clients", fetcher);
+}
+
+export const usePost = () => {
```
Comment (jewelltaylor) [inline florist/app/jobs/hooks.tsx:22]: Very cool! I don't know if this is a common pattern but I liked the way you set this up

**Review by jewelltaylor — APPROVED** [review]
Spent a long time trying to make sense of everything and test out the functionality and everything seems rock solid! Hopefully as I wrap my head around more JS stuff I can provide better feedback :)

File: florist/app/jobs/hooks.tsx (hunk 2 above)
Comment (lotif) [reply florist/app/jobs/hooks.tsx:22]: Yes, I tried multiple ways of doing it and found this in a medium post which appeared to be the most elegant way. I didn't know about hooks before I started working with Next.js (they were not a thing the last time I worked with React in 2017) and it's growing on me as a concept.

**Comment by lotif** [comment]
@jewelltaylor thanks for the review! I realize it is a long PR with many different concepts being introduced but I hope that is the last one of the long ones!

**Comment by lotif** [comment]
Oops I forgot to make the modifications you asked, will push them straight into the main branch as they were very small and only on docstrings.

**Comment by jewelltaylor** [comment]
No worries hahah there was not much to change. Thanks again for the great PR :)

**Pull request merged**